primitives/npos-elections/compact/src/lib.rs

```diff
@@ -239,20 +233,74 @@ fn imports() -> Result<TokenStream2> {
 	}
 }
 
-struct CompactSolutionDef {
+struct SolutionDef {
 	vis: syn::Visibility,
 	ident: syn::Ident,
+	voter_type: syn::Type,
+	target_type: syn::Type,
+	weight_type: syn::Type,
 	count: usize,
+	compact_encoding: bool,
+}
+
+fn check_compact_attr(input: ParseStream) -> bool {
+	let mut attrs = input.call(syn::Attribute::parse_outer).unwrap_or_default();
+	if attrs.len() == 1 {
+		let attr = attrs.pop().expect("Vec with len 1 can be popped.");
+		if attr.path.segments.len() == 1 {
+			let segment = attr.path.segments.first().expect("Vec with len 1 can be popped.");
+			if segment.ident == Ident::new("compact", Span::call_site()) {
+				return true
+			}
+		}
+	}
+	false
```

Comment:
This functions will return false on typos: like if user write `#[compac]` or `#[conpact]` no warning will be shown and encoding won't be compact. I think if attribute is of len 0 then we should return false, otherwise we return true or error.

Reply:
Agree 👍